src/app/facility/analysis/run-analysis/group-analysis/regression-model-selection/model-filter.pipe.ts

```diff
@@ -2,17 +2,27 @@ import { Pipe, PipeTransform } from '@angular/core';
 import { JStatRegressionModel } from 'src/app/models/analysis';
 
 @Pipe({
-    name: 'modelFilter',
-    standalone: false
+  name: 'modelFilter',
+  standalone: false
 })
 export class ModelFilterPipe implements PipeTransform {
 
-  transform(models: Array<JStatRegressionModel>, showInvalid: boolean): Array<JStatRegressionModel> {
-    if (!showInvalid) {
-      return models.filter(model => { return model.isValid });
-    } else {
-      return models;
+  transform(models: Array<JStatRegressionModel>, showInvalid: boolean, filterType: 'invalidModel' | 'failedDataValidation'): Array<JStatRegressionModel> {
```

Comment:
I was thinking for the filter you would pass both of the "showInvalid" and "showFailedValidationModel" booleans and then only need to use the filter once on the array.